src/backend/distributed/connection/placement_connection.c

```diff
@@ -1164,16 +1164,36 @@ ColocatedPlacementsHashCompare(const void *a, const void *b, Size keysize)
 
 
 /*
- * AnyConnectionAccessedPlacements simply checks the number of entries in
- * ConnectionPlacementHash. This is useful to detect whether we're in a
- * distirbuted transaction and already executed at least one command that
- * accessed to a placement.
+ * AnyConnectionModifiedPlacements checks the entries in ConnectionPlacementHash
+ * and determines if we executed a command that modified a placement. This is
+ * useful to detect whether we're in a distributed transaction and already
+ * executed at least one command that modified a placement.
  */
 bool
-AnyConnectionAccessedPlacements(void)
+AnyConnectionModifiedPlacements(void)
```

Comment:
Do we need this change for the PR?

We've not done this in the past because of performance reasons. You have to iterate over the connection hash now.

https://github.com/citusdata/citus/pull/2938#pullrequestreview-285478511

I'd not do it in this PR if it is not strictly necessary

Reply:
I'm guessing we need to change something, since one of the main use cases is:
`WITH a AS (SELECT ... FROM distributed_table ...) SELECT * FROM a ...;` which clearly accesses placements.